requests/exceptions.py

```diff
@@ -25,6 +35,10 @@ def __init__(self, *args, **kwargs):
         super(RequestException, self).__init__(*args, **kwargs)
 
 
+class JSONDecodeError(StandardJSONDecodeError, SimpleJSONDecodeError):
```

Comment:
Our custom exceptions in Requests should ideally fall under a catch-all parent exception, `RequestException`. That allows users to catch everything thrown by Requests.

Reply:
The point of this exception was to catch other `JSONDecodError`s. A `RequestException` would imply something is wrong with the request, when the only thing that went wrong is the json decoding process. This is to indicate that the response text was not json, not that the request itself failed.

Reply:
I made `JSONDecodeError` a subclass of `RequestException` as well.